Formula/syslog-ng.rb

```diff
@@ -0,0 +1,50 @@
+class SyslogNg < Formula
+  desc "Log daemon with advanced processing pipeline and a wide range of I/O methods"
+  homepage "https://www.syslog-ng.com"
+  url "https://github.com/syslog-ng/syslog-ng/releases/download/syslog-ng-3.38.1/syslog-ng-3.38.1.tar.gz"
+  sha256 "5491f686d0b829b69b2e0fc0d66a62f51991aafaee005475bfa38fab399441f7"
+  license all_of: ["LGPL-2.1-or-later", "GPL-2.0-or-later"]
+
+  depends_on "autoconf" => :build
+  depends_on "autoconf-archive" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "glib"
+  depends_on "hiredis"
+  depends_on "ivykis"
+  depends_on "json-c"
+  depends_on "libdbi"
+  depends_on "libmaxminddb"
+  depends_on "libnet"
+  depends_on "librdkafka"
+  depends_on "mongo-c-driver"
+  depends_on "openssl@1.1"
+  depends_on "pcre"
+  depends_on "python@3.11"
+  depends_on "riemann-client"
+
+  uses_from_macos "curl"
+
+  def install
+    sng_python_ver = Formula["python@3.11"].version.major_minor
+
+    system "./configure", *std_configure_args,
+                          "--disable-silent-rules",
+                          "--sbindir=#{bin}",
+                          "--sysconfdir=#{etc}",
+                          "--localstatedir=#{var}",
```

Comment:
Is this not inherited from the prefix?

Reply:
I was told to use `#{HOMEBREW_PREFIX}/var` instead of `#{prefix}/var` (https://github.com/Homebrew/homebrew-core/pull/115410#discussion_r1020181339), so I replaced it as described in the cookbook:
https://docs.brew.sh/Formula-Cookbook#variables-for-directory-locations

Reply:
And for etc?

Reply:
I had to persist user configuration, so I followed the cookbook's suggestion:

> . If after installation you find that to-be-persisted configuration files are not copied but instead symlinked into /usr/local/etc/ from the Cellar, this can often be rectified by passing an appropriate argument to the package’s configure script. That argument will vary depending on a given package’s configure script and/or Makefile, but one example might be: --sysconfdir=#{etc}

Without `--sysconfdir=#{etc}`:
```
lrwxr-xr-x   1 45 Nov 16 13:09 syslog-ng.conf -> ../Cellar/syslog-ng/3.38.1/etc/syslog-ng.conf
```